domain/schema/model.go

```diff
@@ -50,25 +51,130 @@ CREATE TABLE model_config (
 `)
 }
 
-func spacesSchema() schema.Patch {
+func spaceSchema() schema.Patch {
 	return schema.MakePatch(`
-CREATE TABLE provider_spaces (
+CREATE TABLE space (
     uuid            TEXT PRIMARY KEY,
-    name            TEXT
+    name            TEXT NOT NULL,
+    is_public       BOOLEAN
+);
+
+CREATE UNIQUE INDEX idx_spaces_uuid_name
+ON space (name);
+
+CREATE TABLE provider_space (
+    provider_id     TEXT PRIMARY KEY,
+    space_uuid      TEXT NOT NULL,
+    CONSTRAINT      fk_provider_space_space_uuid
+        FOREIGN KEY     (space_uuid)
+        REFERENCES      space(uuid)
+);
+
+CREATE UNIQUE INDEX idx_provider_space_space_uuid
+ON provider_space (space_uuid);
+`)
+}
+
+func subnetSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE subnet (
+    uuid                         TEXT PRIMARY KEY,
+    cidr                         TEXT NOT NULL,
+    vlan_tag                     INT,
+    is_public                    BOOLEAN,
+    space_uuid                   TEXT,
+    subnet_type_uuid             TEXT,
+    CONSTRAINT                   fk_subnets_spaces
+        FOREIGN KEY                  (space_uuid)
+        REFERENCES                   space(uuid)
+    CONSTRAINT                   fk_subnet_types
+        FOREIGN KEY                  (subnet_type_uuid)
+        REFERENCES                   subnet_type(uuid)
+);
+
+CREATE TABLE subnet_type (
+    uuid                         TEXT PRIMARY KEY,
+    name                         TEXT NOT NULL,
+    is_usable                    BOOLEAN,
+    is_space_settable            BOOLEAN
+);
+
+INSERT INTO subnet_type VALUES
+    (0, 'base', true, true),    -- The base (or standard) subnet type. If another subnet is an overlay of a base subnet in fan bridging, then the base subnet is the underlay in fan terminology.
+    (1, 'overlay', false, false),    
+    (2, 'overlay_segment', true, true);
+
+CREATE TABLE subnet_role_definition (
+    uuid                         TEXT PRIMARY KEY,
+    name                         TEXT NOT NULL
 );
 
-CREATE TABLE spaces (
+INSERT INTO subnet_role_definition VALUES
+    (0, 'overlay_of');    -- The subnet is an overlay of other (an underlay) subnet.
+
+CREATE TABLE subnet_type_role_mapping (
+    subnet_type_uuid           TEXT PRIMARY KEY,
+    subnet_role_uuid           TEXT NOT NULL,
+    CONSTRAINT                 fk_subnet_type_uuid
+        FOREIGN KEY                (subnet_type_uuid)
+        REFERENCES                 subnet_type(uuid)
```

Comment:
Here and in other places, you need a comma to separate the constraints.

Reply:
fixed